table/column.go

```diff
@@ -149,7 +150,6 @@ func CastValue(ctx sessionctx.Context, val types.Datum, col *model.ColumnInfo) (
 	sc := ctx.GetSessionVars().StmtCtx
 	casted, err = val.ConvertTo(sc, &col.FieldType)
 	// TODO: make sure all truncate errors are handled by ConvertTo.
-	err = sc.HandleTruncate(err)
```

Comment:
If this is removed, CastValue will always return immediately when meeting truncate error. But there is a lot of logic after this point. Could you add a test case for this case?